src/test/java/com/endava/internship/collections/StudentListTests.java

```diff
@@ -0,0 +1,367 @@
+        studentList.remove(1);
+
+        assertAll(
+                () -> assertThat(studentList.size()).isEqualTo(2),
+                () -> assertThat(studentList).doesNotContain(student1),
+                () -> assertThat(studentList.get(1)).isEqualTo(student2)
+        );
+    }
+
+    @Test
+    public void whenIteratorCurrentPositionIsGreaterThanZero_listIteratorHasPreviousShouldReturnTrue(){
+        studentList.add(student0);
+        studentList.add(student1);
+        studentList.add(student2);
+
+        ListIterator<Student> listIterator = studentList.listIterator();
+        listIterator.nextIndex();
+        assertTrue(listIterator.hasPrevious());
+
+    }
+
+    @Test
+    public void IteratorReturnsPreviousLocation_shouldReturnPreviousIteratorPosition(){
+        studentList.add(student0);
+        studentList.add(student1);
+        studentList.add(student2);
+
+        ListIterator<Student> listIterator = studentList.listIterator();
+        listIterator.nextIndex();
+        assertThat(listIterator.previousIndex()).isEqualTo(0);
+    }
+
+    @Test
+    public void iteratorRemoveAtCurrentPosition_shouldRemoveElementAtThatPosition(){
+        studentList.add(student0);
+        studentList.add(student1);
+        studentList.add(student2);
+
+        ListIterator<Student> listIterator = studentList.listIterator();
+        listIterator.nextIndex();
+        listIterator.remove();
+        assertThat(studentList).doesNotContain(student1);
+    }
+    @Test
+    public void iteratorSetAtCurrentPosition_shouldChangeElementAtCurrentPositionWithAnewOne(){
+        studentList.add(student0);
+        studentList.add(student1);
+        studentList.add(student2);
+
+        ListIterator<Student> listIterator = studentList.listIterator();
+        listIterator.nextIndex();
+        listIterator.set(student4);
+        assertThat(studentList).contains(student4);
+        assertThat(studentList.size()).isEqualTo(3);
+    }
+
+    @Test
+    public void iteratorAddAtCurrentPosition_shouldAddTheNewElementAndIncreaseSize(){
+        studentList.add(student0);
+        studentList.add(student1);
+        studentList.add(student2);
+
+        ListIterator<Student> listIterator = studentList.listIterator();
+        listIterator.add(student4);
+        assertAll(
+                () -> assertThat(studentList).contains(student4),
+                () -> assertThat(studentList).contains(student2),
+                () -> assertThat(studentList).contains(student4),
+                () -> assertThat(studentList).contains(student0),
+                () -> assertThat(studentList.size()).isEqualTo(4)
+        );
+    }
+
+    @Test
+    public void iteratorPrevious_shouldReturnElementAtPreviousPosition(){
+        studentList.add(student0);
+        studentList.add(student1);
+        studentList.add(student2);
+
+        ListIterator<Student> listIterator = studentList.listIterator();
+        listIterator.nextIndex();
```

Comment:
your nextIndex is not working correctly , you are returning the index,
Try to do the same test for ArrayList from collections and see if they are working the same

Reply:
Done(fixed)